### PR TITLE
fix: select pods not in evicted status

### PR DESCRIPTION
**Pull request by vigith**
fixes: #785

### CODE REVIEW

File: pkg/reconciler/vertex/controller.go [hunk 1]
```diff
@@ -181,7 +181,7 @@ func (r *vertexReconciler) reconcile(ctx context.Context, vertex *dfv1.Vertex) (
 		needToCreate := true
 		for existingPodName, existingPod := range existingPods {
 			if strings.HasPrefix(existingPodName, podNamePrefix) {
-				if existingPod.GetAnnotations()[dfv1.KeyHash] == hash {
+				if existingPod.GetAnnotations()[dfv1.KeyHash] == hash && existingPod.Status.Phase != corev1.PodFailed {
```
[the remark this answers is not in the document]
Comment (yhl25) [reply pkg/reconciler/vertex/controller.go]: done.

[the remark this answers is not in the document]
Comment (sarabala1979) [reply pkg/reconciler/vertex/controller.go:184]: So the fix will create pods if the existing pod is in the failed state. Is my understanding correct?  How Will the code handle name for new pods? Will it use a new name or an old name?

[the remark this answers is not in the document]
Comment (vigith) [reply pkg/reconciler/vertex/controller.go:184]: It should be creating a new one `replica-<hash>` where replica should be same but hash should be different.

[the remark this answers is not in the document]
Comment (juliev0) [reply pkg/reconciler/vertex/controller.go:184]: line 220

**Review by juliev0 — APPROVED** [review]
So, you will first locate all Pods in any state that already exist for this Vertex. Then you go through each Pod that needs to exist and:
- if one is found which is failed, you’ll create a new one and delete the old one
- If one is found which is in Pending, Running, Succeeded, or Unknown, you’ll just leave it that way
- If one is not found, you’ll create it

I think that seems good. I considered "unknown" state, but seems reasonable to leave it be just in case since we don't know (apparently often an issue of losing communication with the kubelet)

**Pull request merged**